source/dxgi/dxgi_swapchain.cpp

```diff
@@ -298,7 +304,7 @@ HRESULT STDMETHODCALLTYPE DXGISwapChain::ResizeBuffers(UINT BufferCount, UINT Wi
 
 	runtime_resize();
 
-	return hr;
+	return S_OK;
```

Comment:
This swallows the `DXGI_ERROR_INVALID_CALL` error and does not report it to the application. I have seen applications ignore that error and still work, hence why ReShade has to ignore it too and resize anyway, but it should probably still be reported to the application. Unless this causes problems with UWP games?

Reply:
I suspect the problem being masked here is one caused by the flip presentation model that UWP requires rather than anything UWP specific.

Flip model changes DWM composition behavior to share resource views rather than copy. Views of swapchain buffers are discarded immediately as part of Present (...)'s design.

In other words, any swapchain RenderTargetView that was attached to a device context before presentation is implictly discarded after. ReShade's stateblock code is explicitly re-attaching views that were discarded and only the application itself should be doing that or you will break a lot of resource management.

If this is the problem you can verify it by issuing a non-blocking Present (...) call and trying ResizeBuffers (...) a second time. If that works, you need to re-design runtime::on_present (...)'s stateblock to eliminate these side effects.